### PR TITLE
Handle the case of multiple values for the same header

### PR DESCRIPTION
**Pull request by dluc**
# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

# Description of the change
The proxy was failing to handle responses where a header showed up multiple times

# Motivation for the change
Fixes https://github.com/Azure/reverse-proxy-dotnet/issues/17

---
This change is [Reviewable](https://reviewable.io/reviews/azure/reverse-proxy-dotnet/18)

### CODE REVIEW

File: ProxyAgent/Proxy.cs [hunk 1]
```diff
@@ -114,7 +115,7 @@ public class Proxy : IProxy
             catch (RedirectException e)
             {
                 responseOut.StatusCode = (int) e.StatusCode;
-                responseOut.Headers.Add(LOCATION_HEADER, e.Location);
+                responseOut.Headers[LOCATION_HEADER] = e.Location;
```
Comment (jillcary) [inline ProxyAgent/Proxy.cs]: If there is no LOCATION_HEADER, will this add it?

File: ProxyAgent/Properties/launchSettings.json [hunk 2]
```diff
@@ -12,19 +12,22 @@
...
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
     },
     "ProxyAgent": {
       "commandName": "Project",
-      "launchBrowser": false,
       "launchUrl": "status",
       "environmentVariables": {
+        "REMOTE_ENDPOINT": "https://vm-123456.somecloud.cloudapp.azure.com",
+        "REDIRECT_HTTP_TO_HTTPS": "false",
```
Comment (jillcary) [inline ProxyAgent/Properties/launchSettings.json]: In the README, the default for REDIRECT_HTTP_TO_HTTPS value is true, should this be true also?

Comment (dluc) [reply ProxyAgent/Properties/launchSettings.json]: It makes development easier if the proxy doesn't redirect by default, allowing to test the proxying feature. One can test the redirection feature changing this value of course, but the proxy will not run over SSL, so the traffic won't go through.

File: ProxyAgent/Proxy.cs (hunk 1 above)
Comment (dluc) [reply ProxyAgent/Proxy.cs]: this will replace the header value. The Location header doesn't support multiple values because the browser can follow only one location.

File: ProxyAgent/Properties/launchSettings.json [hunk 3]
```diff
@@ -12,19 +12,22 @@
       "commandName": "IISExpress",
       "launchBrowser": true,
       "environmentVariables": {
-        "REMOTE_ENDPOINT": "http://vm-snjz6.westus.cloudapp.azure.com",
-        "Key": "Value",
+        "REMOTE_ENDPOINT": "https://vm-123456.somecloud.cloudapp.azure.com",
```
Comment (timlaverty) [inline ProxyAgent/Properties/launchSettings.json]: can you verify the old remote endpoint is gone?

File: ProxyAgent/Proxy.cs (hunk 1 above)
Comment (timlaverty) [reply ProxyAgent/Proxy.cs]: so we were previously adding a second location header, yes?

**Comment by dluc** [comment]
I'm adding some tests before merging, will need one more review for approval

File: ProxyAgent/Properties/launchSettings.json (hunk 3 above)
Comment (dluc) [reply ProxyAgent/Properties/launchSettings.json]: yep, verified already, it's gone

File: ProxyAgent/Proxy.cs (hunk 1 above)
Comment (dluc) [reply ProxyAgent/Proxy.cs]: it was technically possible, but not happening

Comment (dluc) [reply ProxyAgent/Proxy.cs]: let me revisit the comment thread:
* answering to Jill: yes, the header is added
* answering to Tim: no, we were not adding a second location header

This code runs only when redirecting a client from HTTP to HTTPS, which happens *before* contacting the remote endpoint, thus there are no headers in `responseOut`.
Technically, one could change the code and inject another middleware upfront of this one, and inject a Location header. In that scenario, it's important to *overwrite* the existing Location header, with this one.
So, to summarize, the previous code was ok, but it's more futureproof taking care of the rare edge case.

**Comment by dluc** [comment]
Tested on a Java deployment
